### PR TITLE
Code Style Standardization: Clang Formatting

### PR DESCRIPTION
**Pull request by Trikzon**
This is an attempt to standardize our code style choices.
It uses clang-format (can be installed on linux with the clang package) to automatically format the code.

To run for all files on linux:
```bash
find -iname *.java | xargs clang-format -i
```

Clang settings can be found in `./.clang-format` and for further information look at [Clang-Format 11 Style Options](https://releases.llvm.org/11.0.0/tools/clang/docs/ClangFormatStyleOptions.html)

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    Priority:        1
+    SortPriority:    0
+  - Regex:           '^<.*'
+    Priority:        2
+    SortPriority:    0
+  - Regex:           '.*'
+    Priority:        3
+    SortPriority:    0
+IncludeIsMainRegex: '([-_](test|unittest))?$'
+IncludeIsMainSourceRegex: ''
```
Comment (Trikzon) [inline .clang-format:66]: This should be removed as it's only used for C++ #include statements

File: common/src/main/java/net/dodogang/plume/Plume.java [hunk 2]
```diff
@@ -19,15 +20,16 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.Items;
 import net.minecraft.world.poi.PointOfInterestType;
+
 import org.apache.logging.log4j.Level;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/Plume.java:26]: I'm not a huge fan of separating imports like this, but I can't find any way to combine different prefix groups in clang-format

File: common/src/main/java/net/dodogang/plume/Plume.java [hunk 3]
```diff
@@ -43,30 +45,30 @@ public static void initialize() {
...
+            ItemGroup itemGroup = new TabbedItemGroup("plume_test", (id) -> {
+                String ns = id.getNamespace();
+                return ImmutableList.of(
+                    TabbedItemGroup.createTab(Blocks.DIRT, ns, "unga"),
+                    TabbedItemGroup.createTab(Blocks.BLACK_CONCRETE, ns, "bunga"));
+            }, () -> new ItemStack(Blocks.STONE));
 
-            BlockRegistryBatch blocks = new BlockRegistryBatch(MOD_ID).setDefaultItemSettings(new Item.Settings().group(itemGroup));
+            BlockRegistryBatch blocks =
+                new BlockRegistryBatch(MOD_ID).setDefaultItemSettings(new Item.Settings().group(itemGroup));
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/Plume.java:56]: I don't like this. I'd rather it be:
```java
BlockRegistryBatch blocks = new BlockRegistryBatch(MOD_ID)
        .setDefaultItemSettings(new Item.Settings().group(itemGroup));
```
I don't know if that's possible tho

File: common/src/main/java/net/dodogang/plume/Plume.java [hunk 4]
```diff
@@ -43,30 +45,30 @@ public static void initialize() {
...
-            TEST_CEILING_PLANT_BLOCK = blocks.add("test_ceiling_plant_block", new CeilingPlantBlock(AbstractBlock.Settings.copy(Blocks.GRASS)));
-            TEST_TALL_CEILING_PLANT_BLOCK = blocks.add("test_tall_ceiling_plant_block", new TallCeilingPlantBlock(AbstractBlock.Settings.copy(Blocks.TALL_GRASS)));
-            TEST_TALLER_PLANT_BLOCK = blocks.add("test_taller_plant_block", new TallerPlantBlock(AbstractBlock.Settings.copy(Blocks.TALL_GRASS)));
+            TEST_BEAM_BLOCK =
+                blocks.add("test_beam_block", new BeamBlock(AbstractBlock.Settings.copy(Blocks.OAK_PLANKS)));
+            TEST_CEILING_PLANT_BLOCK = blocks.add(
+                "test_ceiling_plant_block", new CeilingPlantBlock(AbstractBlock.Settings.copy(Blocks.GRASS)));
+            TEST_TALL_CEILING_PLANT_BLOCK = blocks.add(
+                "test_tall_ceiling_plant_block",
+                new TallCeilingPlantBlock(AbstractBlock.Settings.copy(Blocks.TALL_GRASS)));
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/Plume.java:65]: I wonder if I could make the trailing `);` go onto the next line

File: common/src/main/java/net/dodogang/plume/ash/client/registry/BlockEntityRendererRegistry.java [hunk 5]
```diff
@@ -18,9 +19,7 @@
      */
     @ExpectPlatform
     public static <T extends BlockEntity> void register(
-            BlockEntityType<T> beType,
-            Function<BlockEntityRenderDispatcher, BlockEntityRenderer<T>> renderer
-    ) {
+        BlockEntityType<T> beType, Function<BlockEntityRenderDispatcher, BlockEntityRenderer<T>> renderer) {
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/ash/client/registry/BlockEntityRendererRegistry.java:22]: Idk how I feel about this

File: common/src/main/java/net/dodogang/plume/block/BeamBlock.java [hunk 6]
```diff
@@ -22,33 +22,41 @@
...
+        this.setDefaultState(this.getStateManager()
+                                 .getDefaultState()
+                                 .with(POST, true)
+                                 .with(NORTH, false)
+                                 .with(EAST, false)
+                                 .with(SOUTH, false)
+                                 .with(WEST, false)
+                                 .with(UP, false)
+                                 .with(DOWN, false)
+                                 .with(WATERLOGGED, false));
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/block/BeamBlock.java:59]: I like this. But also, I manually did this. The formatter didn't remove it tho, so I think we might be able to format some of its bad decisions manually

File: common/src/main/java/net/dodogang/plume/block/BeamBlock.java [hunk 7]
```diff
@@ -129,23 +145,20 @@ public boolean isTranslucent(BlockState state, BlockView world, BlockPos pos) {
...
+                map -> beamBlock.getStateManager().getStates().forEach(state -> map.put(state, getStateShape(state))));
         }
 
         private VoxelShape getStateShape(BlockState state) {
             final double size = 4;
-            final VoxelShape baseShape = !state.get(POST)
-                ? Block.createCuboidShape(size, size, size, 16.0D - size, 16.0D - size, 16.0D - size)
-                : Block.createCuboidShape(size, 0.0D, size, 16.0D - size, 16.0D, 16.0D - size);
+            final VoxelShape baseShape =
+                !state.get(POST) ? Block.createCuboidShape(size, size, size, 16.0D - size, 16.0D - size, 16.0D - size)
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/block/BeamBlock.java:160]: ew

File: common/src/main/java/net/dodogang/plume/block/PlumeCraftingTableBlock.java [hunk 8]
```diff
@@ -21,11 +21,9 @@ public PlumeCraftingTableBlock(Settings settings) {
...
-                (syncId, inventory, player) -> new PlumeCraftingScreenHandler(
-                    syncId,
-                    inventory,
-                    ScreenHandlerContext.create(world, pos),
-                    state.getBlock()
-                ), TITLE);
+            (syncId, inventory, player)
+                -> new PlumeCraftingScreenHandler(
+                    syncId, inventory, ScreenHandlerContext.create(world, pos), state.getBlock()),
+            TITLE);
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/block/PlumeCraftingTableBlock.java:27]: I don't like this.

File: common/src/main/java/net/dodogang/plume/block/TallCeilingPlantBlock.java [hunk 9]
```diff
@@ -45,19 +45,28 @@ protected void appendProperties(StateManager.Builder<Block, BlockState> builder)
...
     @Override
-    public BlockState getStateForNeighborUpdate(BlockState state, Direction direction, BlockState newState, WorldAccess world, BlockPos pos, BlockPos posFrom) {
+    public BlockState getStateForNeighborUpdate(
+        BlockState state, Direction direction, BlockState newState, WorldAccess world, BlockPos pos, BlockPos posFrom) {
         DoubleBlockHalf doubleBlockHalf = state.get(HALF);
-        if (direction.getAxis() == Direction.Axis.Y && doubleBlockHalf == DoubleBlockHalf.LOWER == (direction == Direction.UP) && (!newState.isOf(this) || newState.get(HALF) == doubleBlockHalf)) {
+        if (direction.getAxis() == Direction.Axis.Y
+            && doubleBlockHalf == DoubleBlockHalf.LOWER == (direction == Direction.UP)
+            && (!newState.isOf(this) || newState.get(HALF) == doubleBlockHalf))
+        {
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/block/TallCeilingPlantBlock.java:54]: I like this way of doing multi-line if statements

File: common/src/main/java/net/dodogang/plume/block/TallCeilingPlantBlock.java [hunk 10]
```diff
@@ -45,19 +45,28 @@ protected void appendProperties(StateManager.Builder<Block, BlockState> builder)
...
             return Blocks.AIR.getDefaultState();
         } else {
-            return doubleBlockHalf == DoubleBlockHalf.LOWER && direction == Direction.DOWN && !state.canPlaceAt(world, pos) ? Blocks.AIR.getDefaultState() : super.getStateForNeighborUpdate(state, direction, newState, world, pos, posFrom);
+            return doubleBlockHalf == DoubleBlockHalf.LOWER && direction == Direction.DOWN
+                    && !state.canPlaceAt(world, pos)
+                ? Blocks.AIR.getDefaultState()
+                : super.getStateForNeighborUpdate(state, direction, newState, world, pos, posFrom);
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/block/TallCeilingPlantBlock.java:60]: I don't like how it puts the ternary operators back an indent... but idk if that's possible to configure

File: common/src/main/java/net/dodogang/plume/block/TallerPlantBlock.java [hunk 11]
```diff
@@ -26,22 +26,26 @@
...
     @Override
-    public BlockState getStateForNeighborUpdate(BlockState state, Direction direction, BlockState newState, WorldAccess world, BlockPos pos, BlockPos posFrom) {
+    public BlockState getStateForNeighborUpdate(
+        BlockState state, Direction direction, BlockState newState, WorldAccess world, BlockPos pos, BlockPos posFrom) {
         TripleBlockPart part = state.get(PART);
         if (direction == Direction.UP && part != TripleBlockPart.UPPER) {
             Block block = newState.getBlock();
-            if (block != this) return Blocks.AIR.getDefaultState();
+            if (block != this)
+                return Blocks.AIR.getDefaultState();
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/block/TallerPlantBlock.java:44]: This is an config option.

File: common/src/main/java/net/dodogang/plume/client/gui/item_group/ItemGroupTabWidget.java [hunk 12]
```diff
@@ -20,14 +21,15 @@
...
         this.texture = texture;
     }
-    public ItemGroupTabWidget(int x, int y, int selectedTabIndex, TabbedItemGroup tab, CreativeInventoryScreen screen, Identifier texture) {
+    public ItemGroupTabWidget(
+        int x, int y, int selectedTabIndex, TabbedItemGroup tab, CreativeInventoryScreen screen, Identifier texture) {
         this(x - 24, (y + 12) + (selectedTabIndex * 24), tab.getTabs().get(selectedTabIndex), (btn) -> {
             tab.setSelectedTabIndex(selectedTabIndex);
             MinecraftClient.getInstance().openScreen(screen);
-            ((ItemGroupTabWidget) btn).isSelected = true;
+            ((ItemGroupTabWidget)btn).isSelected = true;
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/client/gui/item_group/ItemGroupTabWidget.java:32]: This is a config option.

File: common/src/main/java/net/dodogang/plume/item/item_group/TabbedItemGroup.java [hunk 13]
```diff
@@ -121,7 +122,9 @@ public boolean isInitialized() {
 
     @Override
     public Text getTranslationKey() {
-        return this.selectedTabIndex != 0 ? new TranslatableText("itemGroup." + id.getNamespace(), this.getSelectedItemTab().getTranslationKey()) : super.getTranslationKey();
+        return this.selectedTabIndex != 0
+            ? new TranslatableText("itemGroup." + id.getNamespace(), this.getSelectedItemTab().getTranslationKey())
+            : super.getTranslationKey();
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/item/item_group/TabbedItemGroup.java:127]: I like this.

File: common/src/main/java/net/dodogang/plume/mixin/client/CreativeInventoryScreenMixin.java [hunk 14]
```diff
@@ -62,10 +67,18 @@ public void render(MatrixStack matrixStack, int mouseX, int mouseY, float delta,
...
-    protected void renderTabIcon(MatrixStack matrixStack, ItemGroup itemGroup, CallbackInfo ci) {
+    @Inject(
+        method = "renderTabIcon",
+        at     = @At(
+            value = "INVOKE",
+            target =
+                "Lnet/minecraft/client/gui/screen/ingame/CreativeInventoryScreen;drawTexture(Lnet/minecraft/client/util/math/MatrixStack;IIIIII)V",
+            shift = At.Shift.BEFORE))
+    protected void
+    renderTabIcon(MatrixStack matrixStack, ItemGroup itemGroup, CallbackInfo ci) {
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/mixin/client/CreativeInventoryScreenMixin.java:78]: I don't know why it put the `protected void` on a different line here.

File: common/src/main/java/net/dodogang/plume/mixin/client/CreativeInventoryScreenMixin.java [hunk 15]
```diff
@@ -62,10 +67,18 @@ public void render(MatrixStack matrixStack, int mouseX, int mouseY, float delta,
...
 
-    @Inject(method = "renderTabIcon", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/ingame/CreativeInventoryScreen;drawTexture(Lnet/minecraft/client/util/math/MatrixStack;IIIIII)V", shift = At.Shift.BEFORE))
-    protected void renderTabIcon(MatrixStack matrixStack, ItemGroup itemGroup, CallbackInfo ci) {
+    @Inject(
+        method = "renderTabIcon",
+        at     = @At(
+            value = "INVOKE",
+            target =
+                "Lnet/minecraft/client/gui/screen/ingame/CreativeInventoryScreen;drawTexture(Lnet/minecraft/client/util/math/MatrixStack;IIIIII)V",
+            shift = At.Shift.BEFORE))
```
Comment (Trikzon) [inline common/src/main/java/net/dodogang/plume/mixin/client/CreativeInventoryScreenMixin.java:76]: This is... interesting.

File: forge/src/main/java/net/dodogang/plume/ash/registry/forge/RegistryBatchImpl.java [hunk 16]
```diff
@@ -50,9 +51,11 @@ public RegistryBatchImpl(RegistryKey<Registry<T>> registryKey, String modId) {
 
     @Override
     protected void registerImpl() {
-        IEventBus modEventBus = ModEventBus.getModEventBus(modId).orElseThrow(() -> new IllegalStateException(
-                "Attempted to register BatchedRegister before registering a ModEventBus for modid '" + modId + "'.")
-        );
+        IEventBus modEventBus = ModEventBus.getModEventBus(modId).orElseThrow(
+            ()
+                -> new IllegalStateException(
```
Comment (Trikzon) [inline forge/src/main/java/net/dodogang/plume/ash/registry/forge/RegistryBatchImpl.java:56]: Having the lambda parameters on a different line than the inside of the lambda is ugly. Why does this even exist?

**Comment by Trikzon** [comment]
For the most part, I'm happy with this. I think I can fix the majority of the issues I lined out in my review. Even if some can't be solved, I think having it standardized and a little bad is better than it having a mismatch of styles.

File: common/src/main/java/net/dodogang/plume/Plume.java (hunk 2 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/Plume.java:26]: I don't mind either of the styles, I never really see imports as I just have them minimised :)

File: common/src/main/java/net/dodogang/plume/Plume.java (hunk 3 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/Plume.java:56]: for builders this short (one 'argument') I prefer these as one line, personally

File: common/src/main/java/net/dodogang/plume/Plume.java (hunk 2 above)
Comment (Trikzon) [reply common/src/main/java/net/dodogang/plume/Plume.java:26]: Same

File: common/src/main/java/net/dodogang/plume/Plume.java (hunk 4 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/Plume.java:65]: I hate the newline after the equals aaaaaaaa
also, same with the builder above, this is too short in my opinion to attempt to format nicely, you may as well leave these as single lines

File: common/src/main/java/net/dodogang/plume/ash/client/registry/BlockEntityRendererRegistry.java (hunk 5 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/ash/client/registry/BlockEntityRendererRegistry.java:22]: hey ho guess what my opinion is: single line signatures :)

File: common/src/main/java/net/dodogang/plume/ash/client/registry/BuiltinItemRendererRegistry.java [hunk 17]
```diff
@@ -35,9 +36,12 @@ public static void register(ItemConvertible item, DynamicItemRenderer renderer)
          * @param matrices        the matrix stack
          * @param vertexConsumers the vertex consumer provider
          * @param light           packed lightmap coordinates
-         * @param overlay         the overlay UV passed to {@link net.minecraft.client.render.VertexConsumer#overlay(int)}
+         * @param overlay         the overlay UV passed to {@link
```
Comment (dvitski) [inline common/src/main/java/net/dodogang/plume/ash/client/registry/BuiltinItemRendererRegistry.java:39]: why

File: common/src/main/java/net/dodogang/plume/block/BeamBlock.java [hunk 18]
```diff
@@ -58,17 +66,24 @@ public BeamBlock(AbstractBlock.Settings settings) {
...
+        Boolean up    = this.isConnectable(world, pos.up(), Direction.DOWN);
+        Boolean down  = this.isConnectable(world, pos.down(), Direction.UP);
+
+        return this.getDefaultState()
+            .with(NORTH, north)
+            .with(EAST, east)
+            .with(SOUTH, south)
+            .with(WEST, west)
+            .with(UP, up)
+            .with(DOWN, down);
```
Comment (dvitski) [inline common/src/main/java/net/dodogang/plume/block/BeamBlock.java:81]: `with` indent further?

File: common/src/main/java/net/dodogang/plume/block/BeamBlock.java (hunk 7 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/block/BeamBlock.java:160]: ew

File: common/src/main/java/net/dodogang/plume/block/PlumeCraftingTableBlock.java (hunk 8 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/block/PlumeCraftingTableBlock.java:27]: ew that ruins every principle of lambda styling

File: common/src/main/java/net/dodogang/plume/block/TallCeilingPlantBlock.java (hunk 9 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/block/TallCeilingPlantBlock.java:54]: I can be indifferent but ehhh

[idea](https://www.jetbrains.com/idea/) posted in discord:
```java
@Override
public BlockState getStateForNeighborUpdate(BlockState state, Direction dir, BlockState newState, WorldAccess world, BlockPos pos, BlockPos posFrom) {
    DoubleBlockHalf half = state.get(HALF);
    if (dir.getAxis() == Direction.Axis.Y && (half == DoubleBlockHalf.LOWER && dir == Direction.UP) && (!newState.isOf(this) || newState.get(HALF) == half)) {
        return Blocks.AIR.getDefaultState();
    } else {
        return half == DoubleBlockHalf.LOWER && dir == Direction.DOWN && !state.canPlaceAt(world, pos)
            ? Blocks.AIR.getDefaultState()
            : super.getStateForNeighborUpdate(state, dir, newState, world, pos, posFrom);
    }
}
```

File: common/src/main/java/net/dodogang/plume/block/TallerPlantBlock.java (hunk 11 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/block/TallerPlantBlock.java:44]: imo this should have curly brackets around it

File: common/src/main/java/net/dodogang/plume/client/gui/item_group/ItemGroupTabWidget.java (hunk 12 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/client/gui/item_group/ItemGroupTabWidget.java:32]: I like the space in between :)

File: common/src/main/java/net/dodogang/plume/item/item_group/TabbedItemGroup.java (hunk 13 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/item/item_group/TabbedItemGroup.java:127]: of course.

File: common/src/main/java/net/dodogang/plume/mixin/client/CreativeInventoryScreenMixin.java (hunk 15 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/mixin/client/CreativeInventoryScreenMixin.java:76]: hmm I'm averse

File: common/src/main/java/net/dodogang/plume/mixin/client/CreativeInventoryScreenMixin.java (hunk 14 above)
Comment (dvitski) [reply common/src/main/java/net/dodogang/plume/mixin/client/CreativeInventoryScreenMixin.java:78]: I also find cringe

File: common/src/main/java/net/dodogang/plume/registry/PointOfInterestTypeAppender.java [hunk 19]
```diff
@@ -33,13 +34,12 @@ public static void appendBlockStates(PointOfInterestType poiType, BlockState...
             // A state can only be in one PointOfInterestType, so we throw an error.
             if (previousPoiType != null) {
                 throw Util.throwOrPause(new IllegalStateException(
-                        String.format("%s is defined for too many PointOfInterestTypes.", state)
-                ));
+                    String.format("%s is defined for too many PointOfInterestTypes.", state)));
```
Comment (dvitski) [inline common/src/main/java/net/dodogang/plume/registry/PointOfInterestTypeAppender.java:37]: newline after opening bracket :(
also an issue elsewhere in various places

File: forge/src/main/java/net/dodogang/plume/ash/registry/forge/RegistryBatchImpl.java (hunk 16 above)
Comment (dvitski) [reply forge/src/main/java/net/dodogang/plume/ash/registry/forge/RegistryBatchImpl.java:56]: can I shoot it

**Review by dvitski — CHANGES_REQUESTED** [review]

**Comment by Trikzon** [comment]
I might still mess with this, but I also think that the pr system we've implemented already solves the code styling issues so I might abandon this.

**Comment by dvitski** [comment]
Honestly, I was thinking that this was kind of redundant too :P

**Pull request closed without merging**